app/src/main/java/com/iyehuda/feelslike/ui/map/MapFragment.kt

```diff
@@ -1,37 +1,222 @@
 package com.iyehuda.feelslike.ui.map
 
+import android.Manifest
+import android.content.pm.PackageManager
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.location.Geocoder
 import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.ImageView
 import android.widget.TextView
+import androidx.appcompat.widget.SearchView
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
 import androidx.fragment.app.Fragment
-import androidx.lifecycle.ViewModelProvider
-import com.iyehuda.feelslike.databinding.FragmentMapBinding
+import com.google.android.gms.maps.CameraUpdateFactory
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.OnMapReadyCallback
+import com.google.android.gms.maps.SupportMapFragment
+import com.google.android.gms.maps.model.BitmapDescriptorFactory
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.MarkerOptions
+import com.google.android.material.bottomsheet.BottomSheetDialog
+import com.iyehuda.feelslike.R
+import com.iyehuda.feelslike.model.Post
+import java.io.IOException
+import java.util.Locale
 
-class MapFragment : Fragment() {
-    private var _binding: FragmentMapBinding? = null
-    private val binding get() = _binding!!
+class MapFragment : Fragment(), OnMapReadyCallback {
+    private lateinit var map: GoogleMap
+    private val posts = mutableListOf<Post>()
```

Comment:
Extract state and logic to ViewModel
The fragment code should only deal with UI operations